### PR TITLE
Upgrade version + few improvements

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Motivation 
- https://github.com/operator-framework/operator-sdk-samples/issues/68 
- https://github.com/operator-framework/operator-sdk/issues/1830

required for the following as well:

- https://github.com/operator-framework/getting-started/issues/43
- https://github.com/operator-framework/getting-started/issues/44

## What
- Upgrade go example from 0.6 to operator-sdk version: v0.10.0, commit: ff80b17737a6a0aade663e4827e8af3ab5a21170
- Add makefile in order to be easier to test the example and clarifies how to use it
- Add the new version of olm-catalogue files 
- Change the Readme

## Steps to verify
- Build an image with `operator-sdk build `
- Run `make install` and check that the Memcached pods are deployed and running 

Following the local tests performed with the image `cmacedo/memcached-operator:operator-sdk-samples` 

Screenshot 2019-09-05 at 23 13 03

### CODE REVIEW

**Comment by camilamacedo86** [comment]
HI @AlexNPavel, 

It is done. Note that I added a Makefile and the command  `make test/e2e` which works as an example for its users as can help us to test the tests options in the upper versions.

File: memcached-operator/README.md [hunk 1]
```diff
@@ -53,50 +43,30 @@ $ operator-sdk build $IMAGE
...
-Run the tests that reside in `test/e2e`:
+**NOTE** The `quay.io/example-inc/memcached-operator` is an example. You should build and push the image for your repository. 
 
-```
-$ operator-sdk test local ./test/e2e
-```
+### Installing
 
-To run go-test with verbose and limit to 2 parallel tests:
+Run `make install` and check the operator and the example Memcached deployed and running in the cluster
```
Comment (jmrodri) [inline memcached-operator/README.md]: Run `make install` to install the operator. Check that the operator is running in the cluster, also check that the example Memcached service was deployed.

File: memcached-operator/README.md [hunk 2]
```diff
@@ -53,50 +43,30 @@ $ operator-sdk build $IMAGE
...
-To run go-test with verbose and limit to 2 parallel tests:
+Run `make install` and check the operator and the example Memcached deployed and running in the cluster 
 
-```
-$ operator-sdk test local ./test/e2e --go-test-flags "-v -parallel=2"
-```
+### Uninstalling 
 
-To run the tests from an image in the cluster, first build the tests into your image:
+To uninstall all that was performed in the above step run `make uninstall`.
```
Comment (jmrodri) [inline memcached-operator/README.md]: To uninstall all that was performed, simply run `make uninstall`.

File: memcached-operator/README.md [hunk 3]
```diff
@@ -53,50 +43,30 @@ $ operator-sdk build $IMAGE
...
-$ kubectl create -f deploy/service_account.yaml -n $NAMESPACE
-$ kubectl create -f deploy/role.yaml -n $NAMESPACE
-$ kubectl create -f deploy/role_binding.yaml -n $NAMESPACE
-```
-
-Finally, run the tests:
-```
-$ operator-sdk test cluster $IMAGE --namespace $NAMESPACE --service-account memcached-operator
-```
+**NOTE:** To check what is done in the Make commands see the [Makefile](./Makefile)
```
Comment (jmrodri) [inline memcached-operator/README.md]: I would remove this line entirely. That's the first thing I would do is see what is in the `Makefile` :) Also, I have a way to add `help` to the `Makefile` so when you run `make help` it prints out all the targets and a brief help message with them.

Here's the help from the Ansible Service Broker's Makefile:
```
[jesusr@transam ansible-service-broker{release-1.2}]$ make help
Usage: make <OPTIONS> ... <TARGETS>

Available targets are:

vendor               Install or update project dependencies
broker               Build the broker
build                Build binary from source
generate             regenerate mocks
lint                 Run golint
fmt                  Run go fmt
fmtcheck             Check go formatting
test                 Run unit tests
test-coverage-html   checkout the coverage locally of your tests
ci-test-coverage     CI test coverage, upload to coveralls
vet                  Run go vet
check                Pre-flight checks before creating PR
prep-local           Prepares the local dev environment
build-image          Build a docker image with the broker binary
release              Builds docker container using latest rpm from Copr
clean                Clean up your working environment
really-clean         Really clean up the working environment
deploy               Deploy a built broker docker image to a running cluster
cleanup-ci           Cleanup after ci run
ci                   Run the CI workflow locally
help                 Show this help screen
```

File: memcached-operator/Makefile [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	@echo ... Running NOT in parallel ...
+	- operator-sdk test local ./test/e2e --go-test-flags "-v -parallel=1"
+	@echo ... Running in parallel ...
+	- operator-sdk test local ./test/e2e --go-test-flags "-v -parallel=2
+	@echo ... Running without options/args ...
+	- operator-sdk test local ./test/e2e
+	@echo ... Running with the --debug param ...
+	- operator-sdk test local ./test/e2e --debug
+	@echo ... Running with the --verbose param ...
+	- operator-sdk test local ./test/e2e --verbose
```
Comment (jmrodri) [inline memcached-operator/Makefile]: `Makefile` looks good.

File: memcached-operator/README.md (hunk 1 above)
Comment (camilamacedo86) [reply memcached-operator/README.md]: Terrific suggestion. Done 👍 tks

File: memcached-operator/README.md (hunk 2 above)
Comment (camilamacedo86) [reply memcached-operator/README.md]: 👍

**Comment by theishshah** [comment]
The operator looks good, fan of the new Makefile! Agree with Jesus about the small comments on the Read Me

File: memcached-operator/README.md (hunk 3 above)
Comment (camilamacedo86) [reply memcached-operator/README.md]: It makes sense. Following the help added.  

Screenshot 2019-09-18 at 21 23 30

Comment (camilamacedo86) [reply memcached-operator/README.md]: c/c @theishshah ^

**Comment by camilamacedo86** [comment]
Hi @theishshah and @jmrodri,

Really tks for your review. All suggestions are applied. 
Please, feel free to check and let me know if we can merge it.

Cheers,

**Pull request merged**